hedera-node/src/main/java/com/hedera/services/txns/crypto/CryptoDeleteAllowanceTransitionLogic.java

```diff
@@ -0,0 +1,216 @@
+		final var op = cryptoDeleteAllowanceTxn.getCryptoDeleteAllowance();
+		entitiesChanged.clear();
+		nftsTouched.clear();
+
+		/* --- Use models --- */
+		final Id payerId = Id.fromGrpcAccount(payer);
+		final var payerAccount = accountStore.loadAccount(payerId);
+
+		/* --- Do the business logic --- */
+		deleteCryptoAllowances(op.getCryptoAllowancesList(), payerAccount);
+		deleteFungibleTokenAllowances(op.getTokenAllowancesList(), payerAccount);
+		deleteNftSerials(op.getNftAllowancesList(), payerAccount);
+
+		/* --- Persist the owner accounts and nfts --- */
+		for (final var nft : nftsTouched) {
+			tokenStore.persistNft(nft);
+		}
+		for (final var entry : entitiesChanged.entrySet()) {
+			accountStore.commitAccount(entry.getValue());
+		}
+
+		txnCtx.setStatus(SUCCESS);
+	}
+
+	/**
+	 * Clear spender on the provided nft serials. If the owner is not provided in any allowance,
+	 * considers payer of the transaction as owner while checking if nft is owned by owner.
+	 *
+	 * @param nftAllowances
+	 * 		given nftAllowances
+	 * @param payerAccount payer for the transaction
+	 */
+	private void deleteNftSerials(final List<NftRemoveAllowance> nftAllowances, final Account payerAccount) {
+		if (nftAllowances.isEmpty()) {
+			return;
+		}
+
+		final var nfts = new ArrayList<UniqueToken>();
+		for (var allowance : nftAllowances) {
+			final var serialNums = allowance.getSerialNumbersList();
+			final var tokenId = Id.fromGrpcToken(allowance.getTokenId());
+			final var accountToWipe = fetchOwnerAccount(allowance.getOwner(), payerAccount, accountStore,
+					entitiesChanged);
+			final var token = tokenStore.loadPossiblyPausedToken(tokenId);
+
+			for (var serial : serialNums) {
+				final var nft = tokenStore.loadUniqueToken(tokenId, serial);
+				validateTrue(validOwner(nft, accountToWipe.getId(), token), SENDER_DOES_NOT_OWN_NFT_SERIAL_NO);
+				nft.clearSpender();
+				nfts.add(nft);
+			}
+			nftsTouched.addAll(nfts);
+			nfts.clear();
+		}
+	}
+
+	/**
+	 * For all given token allowance sto be deleted, deletes the token entry from the fungible token allowances
+	 * map on the owner account.If the owner is not provided in any allowance, considers payer of the transaction as
+	 * owner.
+	 *
+	 * @param tokenAllowances
+	 * 		given token allowances
+	 * @param payerAccount
+	 * 		payer for the transaction
+	 */
+	private void deleteFungibleTokenAllowances(final List<TokenRemoveAllowance> tokenAllowances,
+			final Account payerAccount) {
+		if (tokenAllowances.isEmpty()) {
+			return;
+		}
+		for (var allowance : tokenAllowances) {
+			final var owner = allowance.getOwner();
+			final var tokenId = allowance.getTokenId();
+			final var accountToWipe = fetchOwnerAccount(owner, payerAccount, accountStore, entitiesChanged);
+			final var tokensMap = accountToWipe.getMutableFungibleTokenAllowances();
+
+			for (Map.Entry<FcTokenAllowanceId, Long> e : tokensMap.entrySet()) {
+				if (e.getKey().getTokenNum().longValue() == tokenId.getTokenNum()) {
```

Comment:
_(Another comment **definitely** not for this PR)_

It would be pretty awesome if our `semanticPrecheck()` function could return a record like,
```
record DeleteAllowanceValidationResult(
    // The entities linked to this validation result, including the timestamp of the signed state used for the validation; and the latest expiry of any linked entity
    LinkedRefs linkedRefs,
    // If non-null, the failure code for the validation
    @Nullable ResponseCodeEnum failureStatus,
    // If non-null, the updated models to commit to Stores (IF no linked refs changed since the signed state) 
    @Nullable DeleteAllowanceModels updatedModels) {
}
```


Since the [`prefetchProcessor` thread pool](https://github.com/hashgraph/hedera-services/blob/3f69611301677365b19fd1173d8a2be4ec62c466/hedera-node/src/main/java/com/hedera/services/ServicesState.java#L248) will have computed this  `DeleteAllowanceValidationResult` asynchronously, by the time this `TxnAccessor` reaches `handleTransaction()`, we will---usually---have nothing to do except check no `LinkedRefs` changed, then call some `Store` commit methods with the `updatedModels`.

_stares off into distance_ 🌅

Reply:
💯  Definitely Michael. WIll add this to the `TxnAccessor` redesign issue